src/components/TagSelector.vue

```diff
@@ -1,81 +1,96 @@
+  <n-space vertical>
+    <n-select 
+      v-model:value="tagIds" 
       multiple
-      searchable
-    >
-      <template #content="{ value }">
-        <va-chip
-          v-for="tag in value"
-          :key="tag"
-          size="small"
-          :color="tag.hexes?.background"
-          closeable
-          @update:modelValue="deleteTag(tag)"
-        >
-          {{ tag.name }}
-        </va-chip>
-      </template>
-    </va-select>
-  </div>
+      filterable
+      placeholder=""
+      :render-tag="renderTag"
+      :options="options"
+    />
+  </n-space>
 </template>
 
 <script lang="ts">
-import { defineComponent, ref, computed, onMounted, PropType, watch } from "vue";
+import { defineComponent, ref, computed, onMounted, watch, PropType, h } from 'vue'
+import { NSelect, NSpace, NTag, SelectRenderTag } from 'naive-ui';
+import { TagOption } from '@/types/vue';
 import { Task, TaskTag } from "@/types/asana";
-import { TagOption } from "@/types/vue";
 import { useAsanaStore } from "@/store/asana";
 import { usePrefStore } from "@/store/preferences";
+import { Color } from 'csstype';
 
-export default defineComponent ({
+export default defineComponent({
+  components: { NSelect, NSpace },
   props: {
     task: {
       type: Object as PropType<Task>,
       required: true
     }
   },
-  setup(props) {
+  setup (props) {
     const asanaStore = useAsanaStore();
     const prefStore = usePrefStore();
-    
-    const value = ref<TagOption[]>([]);
+    const tagIds = ref<string[]>([]);
 
     const options = computed(() => {
       const tags = asanaStore.allTags;
       return makeTagOption(tags);
     });
 
-    const deleteTag = (tag) => {
-      value.value = value.value.filter((v) => v !== tag)
-    }
-
     onMounted(() => {
-      value.value = makeTagOption(props.task.tags);
-    })
+      tagIds.value = makeTagId(props.task.tags);
+    });
 
-    watch([value], () => {
-      const tagIds = value.value.map((tag) => tag.gid);
-      prefStore.SET_NEW_TAGS(tagIds);
-    })
+    watch([tagIds], () => {
+      prefStore.SET_NEW_TAGS(tagIds.value);
+    });
+
+    const renderTag: SelectRenderTag = ({ option, handleClose }) => {
+      return h(
+        NTag,
+        {
+          closable: true,
+          round: true,
+          themeOverrides: { 
+            color: option.color as Color, 
+            border: option.color as Color,
+            textColor: option.font as Color, 
+            closeColor: option.font as Color,
+          },
+          onClose: (e: MouseEvent) => {
+            e.stopPropagation()
+            handleClose()
+          }
+        },
+        { default: () => option.label }
+      );
+    }
 
     return {
+      tagIds,
       options,
-      value,
-      deleteTag,
+      renderTag,
     }
-  },
+  }
 });
 
+// value prop of component only accepts an array of strings, so need array of tagIds
+function makeTagId(tags: TaskTag[]): string[] { 
+  const tagIds = tags?.map((tag) => {
```

Comment:
The question mark here suggests that the input can be nullable. But the types don't reflect that. Should we update?